include/node.tcc

```diff
@@ -346,6 +347,63 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_velocity_constraints() {
   }
 }
 
+// !Apply absorbing constraints
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::apply_absorbing_constraint(
+    unsigned dir, double delta, double h_min, double a, double b) {
+  bool status = true;
+  try {
+
+    if (dir >= Tdim) {
+      throw std::runtime_error("Direction is out of bounds");
+      status = false;
+    }
+    if (material_ids_.size() != 0) {
+      // Get material id
+      auto mat_id = material_ids_.begin();
+
+      // Extract material properties and displacements
+      double pwave_v = this->property_handle_->property(
+          "wave_velocities", prop_id_, *mat_id, Tdim)(0);
+      double swave_v = this->property_handle_->property(
+          "wave_velocities", prop_id_, *mat_id, Tdim)(1);
+      double density =
+          this->property_handle_->property("density", prop_id_, *mat_id)(0);
+      Eigen::Matrix<double, Tdim, 1> material_displacement =
+          this->property_handle_->property("displacements", prop_id_, *mat_id,
+                                           Tdim);
+
```

Comment:
Based on current implementation, these nodal displacements are weighted by the particle mass. We need to divide by nodal mass. Something simple like
```
material_displacement /= this->mass(*mat_id);
```
will do the trick